### ISSUE TITLE
libuhdm.so should have a numbered SONAME

### ISSUE THREAD

**Issue by yurivict** [chipsalliance/UHDM#992]
Currently the SONAME is ```libuhdm.so```.

But you make incompatible API changes. For example, ```BaseClass::Compare()``` got a new argument in the latest version.
This would break consumers of this library.

SONAME should be something like ``````libuhdm.so.A.B.C```, with numbers A, B, C.

**Comment by timkpaine** [comment]
I agree with this, both UHDM and Surelog should really produce `lib<name>.so.major.minor`, `lib<name>.so.major`, and `lib<name>.so`. Given how “fast and loose” we play with versioning right now, this lets us let some tools move forward without breaking others (especially now that we produce shared libraries).
